lesson01/assignment/script-2.js

```diff
@@ -0,0 +1,108 @@
+// Search the "form", and handle its submit event
+const form = document.querySelector('#first-form');
+form.addEventListener('submit', function (event) {
+    event.preventDefault()
+    // create an array to transform or pass the user data, here an empty array
+    let data = [ 'love' , 'peace', '&', 'harmony']
+   
+    //Target the input field to get data from the user
+    const dataUser = document.querySelector('form .entry');
+
+    // Get the user's data
+    const grabDataUser = dataUser.value;
+
+    // Print the output in the form plus a text, and display the data string
+    document.querySelector('form span').innerHTML = `${'This is what you get '}` + ' " ' + grabDataUser + ' " ' + data.toString(); 
+
+    // Slice method
+    // Only works with words that contains at least 5 characters
+    // Targest "slc1" button, and handle its click event
+    slc1.addEventListener('click', function (e) {
+        e.preventDefault();
+
+        // slice method applied from the user's data
+        let slc = grabDataUser.slice(0, -4);
+        
+        // Print the output
+        document.getElementById('slc').innerHTML =  slc;  
+    })
+})

```

Comment:
This looks like it's missing a few lines. Variables `slc1` and `grabDataUser` are not set before their use, which would cause the whole file to error. I think not enough manual testing was done in the browser, or you would have caught this.